src/dos/drive_local.cpp

```diff
@@ -560,6 +566,36 @@ bool localFile::Close() {
 		fhandle = 0;
 		open = false;
 	};
+
+	if (newtime) {
+		// backport from DOS_PackDate() and DOS_PackTime()
+		struct tm tim = { 0 };
+		tim.tm_sec  = (time&0x1f)*2;
+		tim.tm_min  = (time>>5)&0x3f;
+		tim.tm_hour = (time>>11)&0x1f;
+		tim.tm_mday = date&0x1f;
+		tim.tm_mon  = ((date>>5)&0x0f)-1;
+		tim.tm_year = (date>>9)+1980-1900;
+		//  have the C run-time library code compute whether standard time or daylight saving time is in effect.
+		tim.tm_isdst = -1;
+		// serialize time
+		mktime(&tim);
+
+		struct utimbuf ftim;
+		ftim.actime = ftim.modtime = mktime(&tim);
+
+		char fullname[CROSS_LEN];
+		strcpy(fullname, basedir);
+		strcat(fullname, name);
+//		Dos_SpecoalChar(fullname, true);
+		CROSS_FILENAME(fullname);
+		if (utime(fullname, &ftim)) {
```

Comment:
*Thinking out loud*

As I mentioned in one of earlier comments - `utime` was "obsoleted" by POSIX, quite a long time ago - it should not be used any more. That's unfortunate (especially since I presume we don't need better timestamp precision than 1s… I hope?). It'll probably still work, but we need to test. If we could use C++17 (we can't), then there's [`last_write_time`](https://en.cppreference.com/w/cpp/filesystem/last_write_time)…


On Linux, [`utimensat`](http://man7.org/linux/man-pages/man2/utimensat.2.html) is recommended replacement, but I don't know if it's appropriate in this context… time to investigate, I guess.